client/blocks/app-promo/index.jsx

```diff
@@ -1,55 +1,85 @@
 import userAgent from 'calypso/lib/user-agent';
-
+import { useSelector } from 'calypso/state';
+import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
 import './style.scss';
 
-export const AppPromo = () => {
+export const AppPromo = ( {
+	title = false,
+	iconSize = 29,
+	source = 'calypso-customer-home',
+	subheader = false,
+	className = 'customer-home__card',
+	hasQRCode = false,
+} ) => {
 	const isRtl = useRtl();
 	const translate = useTranslate();
 	const localizeUrl = useLocalizeUrl();
-	const { isDesktop, isiPad, isiPod, isiPhone, isAndroid } = userAgent;
+	const isLoggedIn = useSelector( isUserLoggedIn );
+	const iconWidth = Math.ceil( ( 49 / 29 ) * iconSize );
+
+	const { isDesktop, isiPad, isiPod, isiPhone, isAndroid, isMobile } = userAgent;
 	const isIos = isiPad || isiPod || isiPhone;
 
-	const showIosBadge = ! isAndroid && ! isDesktop;
-	const showAndroidBadge = ! isIos && ! isDesktop;
-	const showBadge = showIosBadge || showAndroidBadge;
+	const showIosBadge = ! isAndroid && ! isDesktop && userAgent;
+	const showAndroidBadge = ! isIos && ! isDesktop && userAgent;
+	const showBadge = ( showIosBadge || showAndroidBadge ) && isMobile;
+	if ( title === false ) {
+		title = translate( 'Get our mobile app' );
+	}
+
+	if ( subheader === false ) {
+		subheader = translate( 'Everything you need to build and grow your site from any device.' );
+	}
+
+	className = classNames( className, 'app-promo' );
 
 	return (
-		<Card className="app-promo customer-home__card">
+		<Card className={ className }>
 			<img
 				className="app-promo__icon"
 				src={ isRtl ? wpToJpImageRtl : wpToJpImage }
-				width="49"
-				height="29"
+				width={ iconWidth }
+				height={ iconSize }
 				alt="WordPress and Jetpack app"
 			/>
 			<div className="app-promo__title">
-				<CardHeading tagName="h2">{ translate( 'Get our mobile app' ) }</CardHeading>
-				<h3 className="app-promo__subheader">
-					{ translate( 'Everything you need to build and grow your site from any device.' ) }
-				</h3>
+				<CardHeading tagName="h2">{ title }</CardHeading>
+				<h3 className="app-promo__subheader">{ subheader }</h3>
 			</div>
-			{ showBadge ? (
+
+			{ showBadge && (
 				<div className="app-promo__app-badges">
+					{ subheader === '' && (
+						<p className="app-promo__app-badges-text">
+							{ translate(
+								'Download the Jetpack Mobile App for the best WordPress.com experiance on the go.'
+							) }
+						</p>
+					) }
 					<AppsBadge
 						storeName={ showIosBadge ? 'ios' : 'android' }
-						utm_source="calypso-customer-home"
+						utm_source={ source }
 					></AppsBadge>
 				</div>
-			) : (
+			) }
+			{ isLoggedIn && ! showBadge && (
 				<Button
 					className="app-promo__link-button is-link"
-					href={ localizeUrl( 'https://apps.wordpress.com/get/?campaign=calypso-customer-home' ) }
+					href={ localizeUrl( `https://apps.wordpress.com/get/?campaign=${ source }` ) }
```

Comment:
Needs a `escapeURIComponent`.
```suggestion
					href={ localizeUrl( `https://apps.wordpress.com/get/?campaign=${ escapeURIComponent ( source ) }` ) }
```